app/views/questionnaire.py

```diff
@@ -576,7 +576,7 @@ def _return_date_answer_value(answer_value):
     return None
 
 
-def _remove_dependent_answers_from_completed_blocks(answer_id, questionnaire_store):
+def _remove_dependent_answers_from_completed_blocks(answer_id, group_instance, questionnaire_store):
```

Comment:
What happens when an answer within a repeating group is dependent on an answer not in that repeating group?

Reply:
What he said ^^^
We should add a test for this other scenario while we are here.

Reply:
We test that in test titles within repeating blocks

Reply:
Thanks, will have a look.

Reply:
If not in a repeating group the group instance is 0 .

Reply:
@AndrewTorrance sorry the test_titles_within_repeating_blocks.json doesn't cover the scenario i'm concerned with. The dependency here is the `confirm-answer` is dependent on the `behalf-of-answer` which are both within the repeating group.

Reply:
Added new test

Reply:
Now we can have a dependency in a repeating group on a non repeating independent variable. Note this was an issue that not only affects titles , but any routing via when .